### PR TITLE
improved grouping tool

### PR DESCRIPTION
**Pull request by bernt-matthias**
- replaces non 'floatable' values to a user supplied default value
- do not strip lines because this removes empty cells at the ends
- added some comments to python code

Would be nice to keep column names .. any ideas? Also I could not run the test manually with planemo since I could not find the test data (results). Can we uncomment the last test?

### CODE REVIEW

File: tools/stats/grouping.xml [hunk 1]
```diff
@@ -19,7 +17,7 @@
     <param name="ignorecase" type="boolean" truevalue="1" falsevalue="0">
       <label>Ignore case while grouping?</label>
     </param>
-    <param name="ignorelines" type="select" display="checkboxes" multiple="True" label="Ignore lines beginning with these characters" help="lines beginning with these are not grouped">
+    <param name="ignorelines" type="select" multiple="True" label="Ignore lines beginning with these characters" help="lines beginning with these are not grouped">
```
Comment (jmchilton) [inline tools/stats/grouping.xml]: Why change checkboxes?

**Comment by jmchilton** [comment]
Gotta bump tool version I think.

File: tools/stats/grouping.xml (hunk 1 above)
Comment (bernt-matthias) [reply tools/stats/grouping.xml]: restored the checkboxes

**Comment by bernt-matthias** [comment]
> Gotta bump tool version I think.

Done

**Comment by mvdbeek** [comment]
We've recently updated the tool for py3 compatibility, can you rebase this against dev ?

**Comment by bernt-matthias** [comment]
hmm. apparently I did something wrong in the git action that was intended to be a rebase. But at least the conflict is gone. The question is if its OK to open a named temporary read only and then let an external program (in this case sort) write in to it. Technically this works -- just a bit nasty.

**Comment by nsoranzo** [comment]
@bernt-matthias I've rebased the PR and I had to squash all your commits in 1, I hope that's fine. Can you double check that the changes look correct to you?

**Comment by bernt-matthias** [comment]
looks good

**Comment by bernt-matthias** [comment]
I guess the failing integration test is unrelated? seems to be fastqc failing.

**Comment by mvdbeek** [comment]
@galaxybot test py3

**Comment by jmchilton** [comment]
Thanks a bunch, looks like a good enhancement!

**Pull request merged**